### ISSUE TITLE
Type error when loading repository data

### ISSUE THREAD

**Issue by juan88** [chrisjbillington/hg-export-tool#8]
Hello!

I'm running this tool with Python 3.5 on Linux Mint and when trying to migrate to git on a bunch of repos I'm receiving this error:

```
Traceback (most recent call last):
  File "exporter.py", line 255, in <module>
    main()
  File "exporter.py", line 251, in main
    BASH
  File "exporter.py", line 175, in process_repo
    amended_commits = fix_branches(hg_repo_copy)
  File "exporter.py", line 92, in fix_branches
    all_heads = get_heads(hg_repo)
  File "exporter.py", line 67, in get_heads
    topo_heads = json.loads(output)
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
```
It seems that there is a type mismatch here exporter.py:67 (I think the same pattern repeats a file lines later).

`
output = subprocess.check_output(cmd, cwd=hg_repo)
topo_heads = json.loads(output)
`

**Comment by chrisjbillington** [comment]
Thanks for the report. This is because `json.loads` in Python < 3.6 requires strings, whereas in newer Pythons bytestrings are allowed as well.

Should be fixed in 7898cec8b004bb9800fdea10660225376dec685f

**Comment by Oxkdjsl** [comment]
@eacherkan-aternity

**Comment by Oxkdjsl** [comment]
Skip to content
chrisjbillington
/
hg-export-tool
Public
Code
Issues
2
Pull requests
Actions
Projects
Security
Insights
Jump to bottom
Type error when loading repository data #8
Closed
juan88 opened this issue on Jun 16, 2020 · 2 comments
Comments
juan88 commented on Jun 16, 2020 • 
Edited 1 time
Hello!

I'm running this tool with Python 3.5 on Linux Mint and when trying to migrate to git on a bunch of repos I'm receiving this error:

Traceback (most recent call last):
  File "exporter.py", line 255, in <module>
    main()
  File "exporter.py", line 251, in main
    BASH
  File "exporter.py", line 175, in process_repo
    amended_commits = fix_branches(hg_repo_copy)
  File "exporter.py", line 92, in fix_branches
    all_heads = get_heads(hg_repo)
  File "exporter.py", line 67, in get_heads
    topo_heads = json.loads(output)
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
It seems that there is a type mismatch here exporter.py:67 (I think the same pattern repeats a file lines later).

output = subprocess.check_output(cmd, cwd=hg_repo) topo_heads = json.loads(output)

chrisjbillington added a commit that referenced this issue on Jun 16, 2020
@chrisjbillington
FIx issue #8 …
7898cec
chrisjbillington commented on Jun 16, 2020
Thanks for the report. This is because json.loads in Python < 3.6 requires strings, whereas in newer Pythons bytestrings are allowed as well.

Should be fixed in 7898cec

@chrisjbillington chrisjbillington closed this as completed on Jun 16, 2020
@juan88 juan88 mentioned this issue on Jun 26, 2020
Program complains of uncommited changes when migrating a repository #9
Open
Oxkdjsl commented 1 minute ago
@eacherkan-aternity


Add a quote, <Ctrl+Shift+.>
Add code, <Ctrl+e>
Add a link, <Ctrl+k>
Directly mention a user or team
Reference an issue, pull request, or discussion
Add saved reply
Slash commands
Leave a comment
Remember, contributions to this repository should follow our GitHub Community Guidelines.
Assignees
No one assigned
Labels
None yet
Projects
None yet
Milestone
No milestone
Development
No branches or pull requests

3 participants
@chrisjbillington
@juan88
@Oxkdjsl
Footer
© 2023 GitHub, Inc.
Footer navigation
Terms
Privacy
Security
Status
Docs
Contact GitHub
Pricing
API
Training
Blog
About
Type error when loading repository data · Issue #8 · chrisjbillington/hg-export-tool

@juan88 juan88 edited on Jun 16, 2020
Hello!

I'm running this tool with Python 3.5 on Linux Mint and when trying to migrate to git on a bunch of repos I'm receiving this error:

Traceback (most recent call last): File "exporter.py", line 255, in <module> main() File "exporter.py", line 251, in main BASH File "exporter.py", line 175, in process_repo amended_commits = fix_branches(hg_repo_copy) File "exporter.py", line 92, in fix_branches all_heads = get_heads(hg_repo) File "exporter.py", line 67, in get_heads topo_heads = json.loads(output) File "/usr/lib/python3.5/json/__init__.py", line 312, in loads s.__class__.__name__)) TypeError: the JSON object must be str, not 'bytes'
It seems that there is a type mismatch here exporter.py:67 (I think the same pattern repeats a file lines later).

Traceback (most recent call last):
  File "exporter.py", line 255, in <module>
    main()
  File "exporter.py", line 251, in main
    BASH
  File "exporter.py", line 175, in process_repo
    amended_commits = fix_branches(hg_repo_copy)
  File "exporter.py", line 92, in fix_branches
    all_heads = get_heads(hg_repo)
  File "exporter.py", line 67, in get_heads
    topo_heads = json.loads(output)
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
It seems that there is a type mismatch here exporter.py:67 (I think the same pattern repeats a file lines later).

output = subprocess.check_output(cmd, cwd=hg_repo) topo_heads = json.loads(output)
